Select Stories Improvements

jcfilben opened the pull request:
<!--- Provide a general summary of the PR in the Title above -->

#### What does this PR do?
Applied the following changes to the Select stories
1. Combined "Themed" into "Custom search" and removed "Themed"
2. Moved "Disabled" out of "Simple"
3. Renamed "Children render" to "Children"
4. Removed "Object options with search and multiple"
5. Simplified "Clear" to only have one Select
6. Removed custom theming from "Create option"
7. Changed "Value label" styling so it doesn't look like a button

#### Where should the reviewer start?

#### What testing has been done on this PR?
storybook

#### How should this be manually tested?
storybook

#### Any background context you want to provide?

#### What are the relevant issues?
Closes #4868 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards Compatible

ShimiSun commented:
We can go ahead and merge this PR. 
@ericsoderberghp, I've put you as the reviewer since you requested the above changes and I thought you may want to confirm your thoughts.

ericsoderberghp commented:
Thanks for the cleanup! Lot's of great improvements already.

A few more suggestions:

![Screen Shot 2021-02-11 at 2 17 41 PM](https://user-images.githubusercontent.com/11637956/107705649-e8efca80-6c73-11eb-8d37-c6e82ee94d13.png)
doesn't look so great as an example

![Screen Shot 2021-02-11 at 2 18 45 PM](https://user-images.githubusercontent.com/11637956/107705745-0f156a80-6c74-11eb-8efa-370cc6c27c53.png)
This also doesn't seem like a good recommendation

I don't know that we still need the "Dark" story. I think that is covered well enough by the "All Components" story where you can toggle dark mode.

"Search" story multiple case closes the drop when selecting an option. "Multiple" story leaves the drop open when selecting an option. It seems like both should be leaving the drop open, like "Multiple" does it.

![Screen Shot 2021-02-11 at 2 22 24 PM](https://user-images.githubusercontent.com/11637956/107706041-919e2a00-6c74-11eb-83ca-60284f40f63f.png)
wondering if we can make this look like something we might recommend. It's a bit blah as is.

jcfilben commented:
Update: 

- Changed Children story to have better contrast. Let me know if there are other changes I should make here

<img width="291" alt="Screen Shot 2021-02-24 at 4 00 58 PM" src="https://user-images.githubusercontent.com/54560994/109077509-7e9a4980-76b9-11eb-9a33-4c7b4b2263ba.png">

- Changed Custom Search story to say 'multiple' when multiple options are selected instead of listing them all

<img width="281" alt="Screen Shot 2021-02-24 at 4 02 15 PM" src="https://user-images.githubusercontent.com/54560994/109077650-ba351380-76b9-11eb-882c-4fa11a6ca256.png">

- Removed Dark story

- Changed Search story multiple case to leave the drop open after selecting an option.

- Changed Value Label story. Let me know if I should change styling to something else

<img width="258" alt="Screen Shot 2021-02-24 at 4 04 35 PM" src="https://user-images.githubusercontent.com/54560994/109077841-ff594580-76b9-11eb-9f39-da4731294376.png">
